ivan-a-kovtun/lab1-BasicIO/user.c

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************/
+/* Files to Include                                                           */
+/******************************************************************************/
+
+#ifdef __XC32
+#include <xc.h>          /* Defines special function registers, CP0 regs  */
+#endif
+
+#include <stdint.h>          /* For uint32_t definition                       */
+#include <stdbool.h>         /* For true/false definition                     */
+#include "user.h"            /* variables/params used by user.c               */
+
+/******************************************************************************/
+/* User Functions                                                             */
+
+/******************************************************************************/
+
+void InitApp(void) {
+    /* Setup analog functionality and port direction */
+
+    // LED output
+    // Disable analog mode for G6
+    ANSELGbits.ANSG6 = 0;
+    // Set direction to output for G6
+    TRISGbits.TRISG6 = 0;
+
+    // Initialize outputs for other LEDs
+    ANSELBbits.ANSB11 = 0;
+    TRISBbits.TRISB11 = 0;
+    
+    ANSELGbits.ANSG15 = 0;
+    TRISGbits.TRISG15 = 0;
+    
+    /* ANSELDbits.ANSD4 = 0;
+       we don't write this string because
+       it is no analog mode for port D
+     */
+    TRISDbits.TRISD4 = 0;
+
+    // Turn on LEDs for testing
+    LD1_PORT_BIT = 1;
+    LD2_PORT_BIT = 1;
+    LD3_PORT_BIT = 1;
+    LD4_PORT_BIT = 1;
+    
+    //test LEDs by this loop
+    /*
+    int j = 0;
+    for (j=0; j<20; j++) {
+        Delay(8000000);
+        LD1_PORT_BIT = ~ LD1_PORT_BIT;
+        LD2_PORT_BIT = ~ LD2_PORT_BIT;
+        LD3_PORT_BIT = ~ LD3_PORT_BIT;
+        LD4_PORT_BIT = ~ LD4_PORT_BIT;
+    }
+    */
+    
+    // Stop program here
+    //while (1);
+    
+    // BTN1 input
+    // Disable analog mode
+    ANSELAbits.ANSA5 = 0;
+    // Set directions to input
+    TRISAbits.TRISA5 = 1;
+
+    // Initialize input for BTN2
+    //ANSELAbits.ANSA4 = 0; As for D4
+    TRISAbits.TRISA4 = 1;
+
+    // Test switches
+    /*
+    while (1) {
+        LD1_PORT_BIT = BTN1_PORT_BIT;
+        LD2_PORT_BIT = BTN2_PORT_BIT;
+    };
+    */
+}
+
+void Delay(int n) {
+    volatile int i;
+    for (i = 0; i < n; i++) {
+    }
+}
+
+void Flash_LED(void) {
+    int delay_count;
+    while (1) {
+        if (1 == PORTAbits.RA5) {
```

Comment:
There's no need to test flag value equality to 1. All non-0 integers are treated as logic 1.
`if (PORTAbits.RA5) {`
Even better solution is to use _bit manipulation_ style testing
`if (PORTA & BIT5) {`